crt0stack/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Create the initial stack frame to start an ELF binary on Linux
+
+#![no_std]
+#![deny(missing_docs)]
+
+extern crate alloc;
+
+pub mod auxv;
+mod stack;
+
+pub use crate::auxv::AuxvEntry;
+
+use crate::auxv::*;
+use alloc::fmt::Debug;
+use alloc::string::String;
+use alloc::vec::Vec;
+use stack::Stack;
+
+/// The error type which is returned from serializing into a byte array.
+///
+/// This type does not support transmission of an error other than that an error
+/// occurred. Any extra information must be arranged to be transmitted through
+/// some other means.
+///
+/// An important thing to remember is that the type `Error` should not be
+/// confused with [`std::io::Error`] or [`std::error::Error`], which you may also
```

Comment:
Would it make sense to diverge from the idiom and give it a more meaningful name?

Reply:
It's the same as `alloc::fmt::Error`
https://doc.rust-lang.org/nightly/src/core/fmt/mod.rs.html#70-96

We might want to extend that..